### PR TITLE
Update linting

### PR DESCRIPTION
**Pull request by henrikstranneheim**
This PR adds/fixes:

### Add

- Pre-commit and config
- Black action
- Isort

### Change

- Update Black and apply it
- Remove flake8

**Review:**
- [ ] Tests executed by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: crunchy/cli/auto_cmd.py [hunk 1]
```diff
@@ -57,7 +57,8 @@ def fastq(ctx, indir, spring_path, first, second, dry_run):
 
     if indir:
         LOG.info(
-            "This will recursively compress and delete fastqs in %s", indir,
+            "This will recursively compress and delete fastqs in %s",
```
Comment (diitaz93) [inline crunchy/cli/auto_cmd.py:60]: is formatting this str out of scope?

Comment (henrikstranneheim) [reply crunchy/cli/auto_cmd.py:60]: I would say so - focusing on the security issue.

File: crunchy/cli/checksum_cmd.py [hunk 2]
```diff
@@ -12,11 +12,8 @@
...
 @click.command()
 @click.argument("infile", type=click.Path(exists=True))
-@click.option(
-    "--algorithm", "-a", type=click.Choice(["md5", "sha1", "sha256"]), default="sha256"
-)
+@click.option("--algorithm", "-a", type=click.Choice(["md5", "sha1", "sha256"]), default="sha256")
 def checksum(infile, algorithm):
-    """Generate the checksum for a file
-    """
+    """Generate the checksum for a file"""
```
Comment (diitaz93) [inline crunchy/cli/checksum_cmd.py:17]:
```suggestion
    """Generate the checksum for a file."""
```

File: crunchy/command.py [hunk 3]
```diff
@@ -46,14 +48,14 @@ def run_command(self, parameters=None):
...
-        )
+        res = subprocess.run(command, check=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
 
         self.stdout = res.stdout.decode("utf-8").rstrip()
         self.stderr = res.stderr.decode("utf-8").rstrip()
         if res.returncode != 0:
-            LOG.critical(f"Call {command} exit with a non zero exit code", )
+            LOG.critical(
+                f"Call {command} exit with a non zero exit code",
+            )
```
Comment (diitaz93) [inline crunchy/command.py:58]: if you remove the comma this will be one line I think, do we want the comma there?

**Review by diitaz93 — APPROVED** [review]
👍

File: crunchy/command.py (hunk 3 above)
Comment (henrikstranneheim) [reply crunchy/command.py]: 👍

**Comment by henrikstranneheim** [comment]
Nothing to deploy

**Pull request merged**